components/aws/sagemaker/common/_utils.py

```diff
@@ -471,11 +471,61 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  endpoint_exists = False
+  if endpoint_name:
+      endpoints = client.list_endpoints(
```

Comment:
1. this need to run in loop to handle case where customer has more than 99 endpoints ?

2. this is an expensive operation, is there a faster way to determine based on describe_endpoint maybe ?

Reply:
Somehow missed the describe_endpoint in the documentation. Changed to using it now.